rust/linux-client/src/main.rs

```diff
@@ -40,15 +40,22 @@ async fn main() -> Result<()> {
         public_key.to_bytes(),
     )?;
 
-    let mut session = Session::connect(
+    let session = Session::connect(
         login,
         private_key,
         None,
-        callbacks,
+        callbacks.clone(),
         max_partition_time,
         tokio::runtime::Handle::current(),
     )
     .unwrap();
+    // TODO: this should be added dynamically
+    session.set_dns(
```

Comment:
These are kind of a "polyfill" until we merge the rest of the network update prs/

Reply:
Can we not call the callbacks here but rather pass the data in that we stored in `Callbacks`?